grails-datastore-gorm/src/main/groovy/org/grails/datastore/gorm/bootstrap/AbstractDatastoreInitializer.groovy

```diff
@@ -296,6 +298,30 @@ abstract class AbstractDatastoreInitializer implements ResourceLoaderAware{
                     datastore = ref("${type}Datastore")
                 }
             }
+            final SoftServiceLoader<Service> services = SoftServiceLoader.load(Service)
+            for (ServiceDefinition<Service> serviceDefinition: services) {
+                if (serviceDefinition.isPresent()) {
+                    final Class<Service> clazz = serviceDefinition.getType()
+                    if (clazz.simpleName.startsWith('$')) {
+                        String serviceClassName = clazz.name - '$' - 'Implementation'
```

Comment:
This will work with all sorts of names other than the standard we are interested in

`if (clazz.simpleName.startsWith('$') && clazz.simpleName.endsWith('Implementation')}`

Then you can just substring it